MatrixSDK/Crypto/Data/Store/MXRealmCryptoStore/MXRealmCryptoStore.m

```diff
@@ -1827,6 +1867,30 @@ + (BOOL)shouldCompactReamDBForUserWithUserId:userId andDevice:(NSString*)deviceI
     return YES;
 }
 
+#pragma mark - readOnly
+
+- (void)setReadOnly:(BOOL)readOnly
+{
+    NSLog(@"[MXRealmCryptoStore] setReadOnly: %@", readOnly ? @"YES" : @"NO");
+    _readOnly = readOnly;
+}
+
+static BOOL copyReadonlyDBNextTime = YES;
```

Comment:
This static variable can lead to unpredictable result.
The app and extension run on their own process. copyReadonlyDBNextTime will be always YES for them at init. But the behavior will be different is different in our tests where we have only one process.

I wonder if we should remove the copyReadonlyDBNextTime, that is a bit hard to understand and add a method `deleteReadOnlyCache` that we use when needed.

Then, we could use the existence of the cache file in https://github.com/matrix-org/matrix-ios-sdk/pull/1106/files?diff=split&w=1#diff-0a4aae3a9f88fcb6a3fc5f1f0d20d2afb3ebbbb55951e55025dfd0c0aa311a26R1640.

Reply:
Refactored.